dissect/target/plugins/os/windows/syscache.py

```diff
@@ -75,7 +75,9 @@ def syscache(self):
                 full_path = None
                 if mft:
                     try:
-                        full_path = self.target.fs.path("\\".join(["sysvol", mft.mft(file_segment).fullpath()]))
+                        path  = mft.ntfs.mft(file_segment).full_path()
```

Comment:
Can you add a unit test to `tests/plugins/os/windows/test_syscache.py`?

Reply:
Current test should cover ``AttributeError`` caused by ``fullpath()`` rename, but I have no clue how to add a test for second issue with MFT record problems, since this plugin gets information from MFT - there is a need to also mock broken MFT record.

Also, currect test from master fails for me with ``PluginError``, I've tried to fix it, with no success:
```
$ tox -- tests/plugins/os/windows/test_syscache.py
...

collected 1 item

tests/plugins/os/windows/test_syscache.py F                                                                                                                                                                                                                                      [100%]

======================================================================================================================================= FAILURES =======================================================================================================================================
_________________________________________________________________________________________________________________________________ test_syscache_plugin _________________________________________________________________________________________________________________________________

self = <Target /home/user/dissect.target/.tox/py3/tmp/test_syscache_plugin0/MockTarget-agy4mufo>, plugin_cls = <class 'dissect.target.plugins.os.windows.syscache.SyscachePlugin'>, check_compatible = True

    def add_plugin(
        self,
        plugin_cls: Union[plugin.Plugin, type[plugin.Plugin]],
        check_compatible: bool = True,
    ) -> plugin.Plugin:
        """Add and register a plugin by class.

        Args:
            plugin_cls: The plugin to add and register, this can either be a class or instance. When this is a class,
                        it will be instantiated.
            check_compatible: A flag that determines if we check whether the plugin is compatible with the ``Target``.

        Returns:
            The ``plugin_cls`` instance.

        Raises:
            UnsupportedPluginError: Raised when plugins were found, but they were incompatible
            PluginError: Raised when any other exception occurs while trying to load the plugin.
        """
        self.log.debug("Adding plugin: %s", plugin_cls)

        if not isinstance(plugin_cls, plugin.Plugin):
            try:
>               p = plugin_cls(self)

dissect/target/target.py:534:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
dissect/target/plugins/os/windows/syscache.py:37: in __init__
    self.hive.add(regutil.RegfHive(fpath))
dissect/target/helpers/regutil.py:629: in __init__
    self.hive = regf.RegistryHive(fh)
.tox/py3/lib/python3.11/site-packages/dissect/regf/regf.py:42: in __init__
    self.header = c_regf.REGF_HEADER(data)
.tox/py3/lib/python3.11/site-packages/dissect/cstruct/types/structure.py:79: in __call__
    return super().__call__(*args, **kwargs)
.tox/py3/lib/python3.11/site-packages/dissect/cstruct/types/base.py:47: in __call__
    return cls.reads(stream)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

cls = <class 'dissect.cstruct.types.structure.REGF_HEADER'>, data = b'version https://git-lfs.github.com/spec/v1\noid sha256:b425cae888e491586ae6e9bc22e856b339af41703e374105ee711e2eacbbdff7\nsize 786432\n'

    def reads(cls, data: bytes) -> BaseType:
        """Parse the given data from a bytes-like object.

        Args:
            data: Bytes-like object to parse.

        Returns:
            The parsed value of this type.
        """
>       return cls._read(BytesIO(data))
E       EOFError

.tox/py3/lib/python3.11/site-packages/dissect/cstruct/types/base.py:75: EOFError

The above exception was the direct cause of the following exception:

target_win = <Target /home/user/dissect.target/.tox/py3/tmp/test_syscache_plugin0/MockTarget-agy4mufo>, fs_win = <VirtualFilesystem>

    def test_syscache_plugin(target_win, fs_win):
        syscache_file = absolute_path("_data/plugins/os/windows/syscache/Syscache.hve")
        fs_win.map_file("System Volume Information/Syscache.hve", syscache_file)

>       target_win.add_plugin(SyscachePlugin)

tests/plugins/os/windows/test_syscache.py:9:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <Target /home/user/dissect.target/.tox/py3/tmp/test_syscache_plugin0/MockTarget-agy4mufo>, plugin_cls = <class 'dissect.target.plugins.os.windows.syscache.SyscachePlugin'>, check_compatible = True

    def add_plugin(
        self,
        plugin_cls: Union[plugin.Plugin, type[plugin.Plugin]],
        check_compatible: bool = True,
    ) -> plugin.Plugin:
        """Add and register a plugin by class.

        Args:
            plugin_cls: The plugin to add and register, this can either be a class or instance. When this is a class,
                        it will be instantiated.
            check_compatible: A flag that determines if we check whether the plugin is compatible with the ``Target``.

        Returns:
            The ``plugin_cls`` instance.

        Raises:
            UnsupportedPluginError: Raised when plugins were found, but they were incompatible
            PluginError: Raised when any other exception occurs while trying to load the plugin.
        """
        self.log.debug("Adding plugin: %s", plugin_cls)

        if not isinstance(plugin_cls, plugin.Plugin):
            try:
                p = plugin_cls(self)
            except PluginError:
                raise
            except Exception as e:
>               raise PluginError(f"An exception occurred while trying to initialize a plugin: {plugin_cls}", cause=e)
E               dissect.target.exceptions.PluginError: An exception occurred while trying to initialize a plugin: <class 'dissect.target.plugins.os.windows.syscache.SyscachePlugin'>

dissect/target/target.py:538: PluginError
---------------------------------------------------------------------------------------------------------------------------------- Captured log setup ----------------------------------------------------------------------------------------------------------------------------------
WARNING  dissect.target.target:target.py:92 <Target None>: Error loading config file: None
=============================================================================================================================== short test summary info ================================================================================================================================
FAILED tests/plugins/os/windows/test_syscache.py::test_syscache_plugin - dissect.target.exceptions.PluginError: An exception occurred while trying to initialize a plugin: <class 'dissect.target.plugins.os.windows.syscache.SyscachePlugin'>
================================================================================================================================== 1 failed in 0.26s ===================================================================================================================================
py3: exit 1 (0.54 seconds) /home/user/dissect.target> pytest --basetemp=/home/user/dissect.target/.tox/py3/tmp tests/plugins/os/windows/test_syscache.py pid=3365
py3: FAIL ✖ in 3.59 seconds
pypy3: skipped because could not find python interpreter with spec(s): pypy3
.pkg: _exit> python /home/user/.local/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta
  lint: FAIL code 1 (1.56=setup[0.04]+cmd[1.52] seconds)
  py3: FAIL code 1 (3.59=setup[3.05]+cmd[0.54] seconds)
  pypy3: SKIP (0.25 seconds)
  evaluation failed :( (5.47 seconds)

Reply:
@JSCU-CNI Ah, my bad, missed that part, thanks! All good now.

Current test seems to not hit affected code, I'll come up with a new test later